crates/compilers/src/compile/project.rs

```diff
@@ -481,10 +514,18 @@ impl<L: Language, S: CompilerSettings> CompilerSources<'_, L, S> {
 
                 input.strip_prefix(project.paths.root.as_path());
 
+                if let Some(preprocessor) = preprocessor.as_ref() {
+                    input =
+                        preprocessor.preprocess(&project.compiler, input, &project.paths, mocks)?;
```

Comment:
take by input reference?

Reply:
yep, changed in https://github.com/foundry-rs/compilers/pull/252/commits/1a3de9f9740cbdbd8853639468d6e06d5f3ce39c